bom/build.gradle

```diff
@@ -1,40 +1,31 @@
 plugins {
+  id 'java-platform'
   id "maven-publish"
 }
 
 description = 'gRPC: BOM'
 
+gradle.projectsEvaluated {
+  def projectsToInclude = rootProject.subprojects.findAll {
```

Comment:
We also need to exclude `grpc-compiler`, as it is added manually in withXml. Right now protoc-gen-grpc-java is in the bom twice (when `-PskipCodegen=false`), one with <type>pom</type> and one without.

(We could keep it here and change withXml to _modify_ the xml for protoc-gen-grpc-java. We do something like that [in netty-shaded](https://github.com/grpc/grpc-java/blob/b1bc0a9d240c2cc6ccf9cf02543c9d9118132c14/netty/shaded/build.gradle#L121).)